crates/wit-parser/src/ast/resolve.rs

```diff
@@ -563,11 +566,21 @@ impl<'a> Resolver<'a> {
             match field {
                 ast::InterfaceItem::Value(value) => match &value.kind {
                     ValueKind::Func(func) => {
-                        self.define_interface_name(&value.name, TypeOrItem::Item("function"))?;
-                        let func = self.resolve_function(&value.docs, value.name.name, func)?;
-                        let prev = self.interfaces[interface_id]
-                            .functions
-                            .insert(value.name.name.to_string(), func);
+                        if !matches!(value.name, VariableId::Wildcard(_)) {
+                            self.define_interface_name(
+                                value.name(),
+                                value.span(),
+                                TypeOrItem::Item("function"),
+                            )?;
+                        }
+                        let func = self.resolve_function(&value.docs, value.name(), func)?;
+                        let prev = if matches!(value.name, VariableId::Wildcard(_)) {
+                            self.interfaces[interface_id].wildcard.replace(func)
```

Comment:
I think we'll want to handle `prev.is_some()` (indicating multiple `*` appears textually) here and return a concrete syntax error positioned at the span of the wildcard.